src/v/kafka/server/handlers/fetch.cc

```diff
@@ -951,6 +951,10 @@ bool update_fetch_partition(
         include = true;
         partition.last_stable_offset = model::offset(resp.last_stable_offset);
     }
+    if (partition.start_offset != resp.log_start_offset) {
```

Comment:
nice find..

q: what are repercussions of this bug? If we miss this and the subsequent request tries to fetch from an evicted offset , I see the reader throws a runtime error, is there more to it? Just making sure I understood it right.

Reply:
without this a consumer will not be able to see the offset updates. I am not sure if there are any major issues, but i wanted to be consistent.